eng/testing/tests.targets

```diff
@@ -136,6 +132,16 @@
       <RunScriptOutputPath>$([MSBuild]::NormalizePath('$(RunScriptOutputDirectory)', '$(RunScriptOutputName)'))</RunScriptOutputPath>
     </PropertyGroup>
 
+    <ItemGroup Condition="'$(IsXUnitLogCheckerSupported)' == 'true' and
+                              '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                              $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '$(NETCoreAppCurrentVersion)'))">
+      <SetScriptCommands Condition="'$(TargetOS)' == 'windows'" Include="set __TestArchitecture=$(TargetArchitecture)" />
+      <SetScriptCommands Condition="'$(TargetOS)' == 'windows'" Include="set __IsXUnitLogCheckerSupported=1" />
+      
+      <SetScriptCommands Condition="'$(TargetOS)' != 'windows'" Include="export __TestArchitecture=$(TargetArchitecture)" />
+      <SetScriptCommands Condition="'$(TargetOS)' != 'windows'" Include="export __IsXUnitLogCheckerSupported=1" />
+    </ItemGroup>
```

Comment:
What's the reason that you moved this block into the target? I would prefer it would stay outside of it. The target itself just calculates inputs and outputs and invokes the msbuild task while this block is proprietary to the xunit log checker.

Reply:
Because I noticed RunScriptCommands is inside the target, and it is consumed exactly like SetScriptCommands. So I wanted to keep both consistent.

Reply:
This is what I mean: https://github.com/dotnet/runtime/pull/95533/files#diff-ea069e899aadcfcf3dcabecd88dbd80e12d6cb331de055fba7f49c9ef9a329adR98-R128

Also, notice how it collects everything into a property, then puts it in an item.

Reply:
I think you refer to L124/L129. That is just putting the RunScriptCommand property into an item, so just build infrastructure for task inputs. The whole target is a general purpose target, not specific to a custom command. Please revert this change and put this back outside of the target.

(FWIW we chose a property back then because the majority of use cases only required a single command. We probably should have just used the item instead.)